src/core/IPosisProcess.d.ts

```diff
@@ -3,6 +3,8 @@ interface IPosisProcess {
     imageName: string; // image name (maps to constructor)
     id: PosisPID; // ID
     parentId: PosisPID; // Parent ID
-    log: IPosisLogger; // Logger 
+    log: IPosisLogger; // Logger
+    // For querying extension interfaces (instead of tying ourselves to "levels")
+    queryPosisInterface<T extends keyof PosisInterfaces>(interfaceId: T): PosisInterfaces[T] | undefined;
```

Comment:
I guess you're talking about the types here? In my tests it seemed to work fine, if you have a case where it doesn't, let me know.